supabase/seed.sql

```diff
@@ -9,7 +9,166 @@ values
+    ('Gabriel', 'Nelson', 'gabriel.nelson@example.com', now()),
+    ('Chloe', 'Carter', 'chloe.carter@example.com', now());
+
+  INSERT INTO clients (first_name, last_name, phone, email, zip_code)
+  VALUES
+    ('Ella', 'Hill', '(345) 678-9012', 'ella.hill@example.com', '40404'),
+    ('John', 'Doe', '(123) 456-7890', 'john.doe@example.com', '10001'),
+    ('Jane', 'Smith', '(234) 567-8901', 'jane.smith@example.com', '10002'),
+    ('Michael', 'Brown', '(345) 678-9012', 'michael.brown@example.com', '10003'),
+    ('Emily', 'Davis', '(456) 789-0123', 'emily.davis@example.com', '10004'),
+    ('Matthew', 'Johnson', '(567) 890-1234', 'matthew.johnson@example.com', '10005'),
+    ('Sophia', 'Miller', '(678) 901-2345', 'sophia.miller@example.com', '10006'),
+    ('James', 'Wilson', '(789) 012-3456', 'james.wilson@example.com', '10007'),
+    ('Olivia', 'Moore', '(890) 123-4567', 'olivia.moore@example.com', '10008'),
+    ('William', 'Taylor', '(901) 234-5678', 'william.taylor@example.com', '10009'),
+    ('Isabella', 'Anderson', '(012) 345-6789', 'isabella.anderson@example.com', '10010'),
+    ('Alexander', 'Thomas', '(123) 456-7890', 'alexander.thomas@example.com', '10011'),
+    ('Mia', 'Jackson', '(234) 567-8901', 'mia.jackson@example.com', '10012'),
+    ('Daniel', 'White', '(345) 678-9012', 'daniel.white@example.com', '10013'),
+    ('Charlotte', 'Harris', '(456) 789-0123', 'charlotte.harris@example.com', '10014'),
+    ('Benjamin', 'Martin', '(567) 890-1234', 'benjamin.martin@example.com', '10015'),
+    ('Amelia', 'Thompson', '(678) 901-2345', 'amelia.thompson@example.com', '10016'),
+    ('Lucas', 'Garcia', '(789) 012-3456', 'lucas.garcia@example.com', '10017'),
+    ('Harper', 'Martinez', '(890) 123-4567', 'harper.martinez@example.com', '10018'),
+    ('Henry', 'Rodriguez', '(901) 234-5678', 'henry.rodriguez@example.com', '10019'),
+    ('Evelyn', 'Martinez', '(012) 345-6789', 'evelyn.martinez@example.com', '10020'),
+    ('Sebastian', 'Lopez', '(123) 456-7890', 'sebastian.lopez@example.com', '10021'),
+    ('Avery', 'Gonzalez', '(234) 567-8901', 'avery.gonzalez@example.com', '10022'),
+    ('Jack', 'Perez', '(345) 678-9012', 'jack.perez@example.com', '10023'),
+    ('Ella', 'Hill', '(456) 789-0123', 'ella.hill2@example.com', '10024'),
+    ('Aiden', 'Scott', '(567) 890-1234', 'aiden.scott@example.com', '10025'),
+    ('Abigail', 'Green', '(678) 901-2345', 'abigail.green@example.com', '10026'),
+    ('Owen', 'Adams', '(789) 012-3456', 'owen.adams@example.com', '10027'),
+    ('Scarlett', 'Baker', '(890) 123-4567', 'scarlett.baker@example.com', '10028'),
+    ('Gabriel', 'Nelson', '(901) 234-5678', 'gabriel.nelson@example.com', '10029'),
+    ('Chloe', 'Carter', '(012) 345-6789', 'chloe.carter@example.com', '10030');
+
+  INSERT INTO pets (name, age, weight, client_id, species)
+  VALUES
+  ('Max', 2, 20, (SELECT id from clients ORDER BY RANDOM() LIMIT 1), (SELECT id FROM app_constants WHERE type = 'species' ORDER BY RANDOM() LIMIT 1)),
```

Comment:
Is there a cleaner way to do this? Maybe a SQL loop?

Reply:
Also I don't like using random in test data, could create some intermittent errors

Reply:
I did some research, we have some options
**1. use supabase api to generate the statements in python then generate seed file** 
the draw back is that we have to put the python into ci/cd before the migration script is executed to make sure seed file is updated.
I dislike this option because it adds extra steps and complexity

**2. use cross join and have something like this**
```sql
INSERT INTO pets (name, age, weight, client_id, species)
SELECT 
  pet_info.name,
  pet_info.age,
  pet_info.weight,
  client_info.id as client_id,
  species.id as species_id
FROM ( 
  VALUES
  ('Max',2, 20),
  ('Sadie', 3, 24),
  ('Gracie', 5, 19),
  ('Roxy', 1, 23)
) AS pet_info(name, age, weight)
CROSS JOIN (
  SELECT id 
  FROM clients
) AS client_info
CROSS JOIN (
  SELECT id 
  FROM app_constants
  WHERE type = 'species'
) AS species;

```

which will produce results similar to this

![image](https://github.com/user-attachments/assets/1e1e891f-fc38-4c64-a79f-184b9175b943)

which we don't want because the test data will be just a bunch of duplicates and repeated info

**3. hand picking uuid values from diff tables to construct pets table (as an example) which takes an owner id and species id**
the problem with this is that the insert statement would still be pretty messy because uuid are long, and it's worse than the current solution because its not as readable and it's harder to tell where the values are coming from.

the nice thing about using this current solution is that you can clearly see where each column value is coming from

also the random ordering is not actually generating a random value to be put into the table, its picking a random existing row from another table. I'd argue this is very predictable and not random at all because we know the entire set of possible results which are all real and not randomly generated.


I've attached the cross join solution so that it doesn't get lost incase we want to go forward with it but personally I think current solution should be kept
[seed.sql.txt](https://github.com/user-attachments/files/16277750/seed.sql.txt)

Reply:
discussed further with @DillonOLeary  on office hours, decided that we will adopt snaplet for seeding, I'll work on it after finishing my sprint-6 tasks